app/models/services/tumblr.rb

```diff
@@ -16,16 +16,21 @@ def consumer_secret
   def post(post, url='')
     body = build_tumblr_post(post, url)
     user_info = JSON.parse(client.get("/v2/user/info").body)
-    blogs = user_info["response"]["user"]["blogs"].map { |blog| URI.parse(blog['url']) }
+    blogs = user_info["response"]["user"]["blogs"]
+    primaryblog = blogs.find(blogs[0]) { |blog| blog["primary"] }
+    if !primaryblog
```

Comment:
Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.